### PR TITLE
proper usage of async await and remove hacky await setState

### PR DESCRIPTION
**Pull request by ellamurii**

### CODE REVIEW

File: src/screens/App.js [hunk 1]
```diff
@@ -46,20 +45,23 @@ class App extends Component {
...
+      const isValidParams = !!(country && article && searchString);
+      this.setState({
+        isFilterComplete: isFirstLoadFilter || isValidParams
+      });
+
+      isValidParams && this.searchNews();
+    };
 
-    isValidParams && this.searchNews();
+    this.setState({ [key]: value }, searchNews);
```
Comment (janjanarnaldo) [inline src/screens/App.js]: Are you setting searchNews function to state?

File: src/screens/App.js [hunk 2]
```diff
@@ -46,20 +45,23 @@ class App extends Component {
...
-      isFilterComplete: isFirstLoadFilter || isValidParams
-    });
+    const searchNews = () => {
+      const { country, article, searchString } = this.state;
+      const isValidParams = !!(country && article && searchString);
+      this.setState({
+        isFilterComplete: isFirstLoadFilter || isValidParams
+      });
+
+      isValidParams && this.searchNews();
```
Comment (janjanarnaldo) [inline src/screens/App.js]: I think the const searchNews  is confusing with this.searchNews

File: src/screens/App.js [hunk 3]
```diff
@@ -30,10 +30,9 @@ class App extends Component {
         country: article === "everything" ? null : country,
         q: searchString
       };
-      await newsApi.get(`/${article}`, { params }).then(({ data }) => {
-        this.setState({
-          searchResults: data.articles
-        });
+      const searchResults = await newsApi.get(`/${article}`, { params });
+      this.setState({
+        searchResults: searchResults.data.articles
```
Comment (janjanarnaldo) [inline src/screens/App.js]: 🎖 🙇‍♀

File: src/screens/App.js (hunk 1 above)
Comment (ellamurii) [reply src/screens/App.js]: it's a callback said the docs hahaha
**setState(updater, [callback])**
https://reactjs.org/docs/react-component.html#setstate

Comment (janjanarnaldo) [reply src/screens/App.js]: oh it's the 2nd arg lol, let me see.

**Pull request closed without merging**